test/IntegrationTests/test_getallchats.py

```diff
@@ -0,0 +1,633 @@
+
+    def test_large_page_size(self):
+        """Test case 11: Large Page Size"""
+        logger.info("=== Starting Test Case 11: Large Page Size ===")
+
+        # Create an initial chat and get the chat_id
+        initial_chat_data = {
+            "customer_guid": self.valid_customer_guid,
+            "question": "You will get the correct answer once AI is integrated."
+        }
+        chat_url = f"{self.BASE_URL}/chat"
+        chat_response = requests.post(chat_url, json=initial_chat_data)
+        self.assertEqual(chat_response.status_code, 200, "Failed to create initial chat.")
+        self.valid_chat_id = chat_response.json().get("chat_id")
+        logger.info(f"Created initial chat with chat_id: {self.valid_chat_id}")
+
+        # Add multiple chat messages to the chat
+        messages_to_add = [
+            "Message 1: You will get the correct answer once AI is integrated.",
+            "Message 2: You will get the correct answer once AI is integrated.",
+            "Message 3: You will get the correct answer once AI is integrated.",
+            "Message 4: You will get the correct answer once AI is integrated.",
+            "Message 5: You will get the correct answer once AI is integrated.",
+        ]
+
+        for message in messages_to_add:
+            message_data = {
+                "customer_guid": self.valid_customer_guid,
+                "chat_id": self.valid_chat_id,  # Include the chat_id here
+                "question": message  # Ensure this key matches what the API expects
+            }
+            logger.info(f"INPUT: Creating message with data: {str(message_data)}")
+            message_response = requests.post(chat_url, json=message_data)
+            logger.info(f"OUTPUT: Message creation response status: {message_response.status_code}")
+            self.assertEqual(message_response.status_code, 200, "Failed to create test message.")
+
+        # Now retrieve all chats using the correct chat_id
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "chat_id": self.valid_chat_id,  # Use the chat_id from the initial chat creation
+            "page": 1,
+            "page_size": 1000
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        # Check for successful response
+        self.assertEqual(response.status_code, 200, "Failed to retrieve chats.")
+
+        response_data = response.json()
+
+        # Check that 'messages' is a list
+        messages = response_data.get('messages')
+        self.assertIsInstance(messages, list, "Expected 'messages' to be a list.")
+
+        # Verify that all added messages are in the response
+        for expected_message in messages_to_add:
```

Comment:
We should also make sure, each request question should have a corresponding answer

Reply:
done sir